acquia_lift.admin.wizard.inc

```diff
@@ -643,3 +807,191 @@ function _acquia_lift_personalize_campaign_wizard_variations_element(&$form, &$f
+      );
+      $num_contexts++;
+    }
+  }
+  $form_state['num_contexts_audience'][$rule_parent_identifier] = count($mappings);
+
+  foreach ($mappings as $delta => $mapping) {
+    $form['mapping']['contexts'][$delta] = personalize_explicit_targeting_mapping_element($mapping, $targeting_values, $delta);
+
+    // Add a "remove" button for this context.
+    // NOTE: ajax.js expects the ID of the element to match the element's name
+    // even when a different selector is passed.
+    $form['mapping']['contexts'][$delta]['remove'] = array(
+      '#prefix' => '<div class="acquia-lift-remove-context">',
+      '#suffix' => '</div>',
+      '#type' => 'submit',
+      '#tag' => 'button',
+      '#text' => t('Remove'),
+      '#value' => 'remove_' . $delta,
+      '#theme_wrappers' => array('personalize_html_tag'),
+      '#attributes' => array(
+        'class' => array('personalize-delete-context', 'form-submit',),
+        'title' => t('Delete this context.'),
+        'id' => 'edit-audiences-' . $rule_parent_identifier . '-details-mapping-contexts-' . $delta . '-remove',
+      ),
+      '#submit' => array('acquia_lift_personalize_campaign_wizard_targeting_remove'),
+      '#ajax' => array(
+        'callback' => 'acquia_lift_personalize_campaign_wizard_targeting_ajax',
+        'wrapper' => $main_wrapper_id,
+        'effect' => 'fade',
+      ),
+    );
+  }
+  // Create an "add new context" link.
+  $form['add_new'] = array(
+    '#prefix' => '<span class="personalize-add-link-prefix"></span>',
+    '#type' => 'submit',
+    '#tag' => 'button',
+    '#text' => t('Add context'),
+    '#value' => 'add_context_' . $rule_parent_identifier,
+    '#theme_wrappers' => array('personalize_html_tag'),
+    '#submit' => array('acquia_lift_personalize_campaign_wizard_targeting_add'),
+    '#attributes' => array(
+      'class' => array('personalize-add-link'),
+      'title' => t('Click here to add more contexts.'),
+      'id' => 'edit-audiences-' . $rule_parent_identifier . '-details-add-new'
+    ),
+    '#ajax' => array(
+      'callback' => 'acquia_lift_personalize_campaign_wizard_targeting_ajax',
+      'wrapper' => $main_wrapper_id,
+      'effect' => 'fade',
+    ),
+  );
+  $form['strategies'] = array(
+    '#type' => 'container',
+  );
+
+  // Add radio buttons so the user can select how multiple features for an option
+  // should be treated.
+  $default_strategy = 'OR';
+  $form['strategy'] = array(
+    '#type' => 'select',
+    '#multiple' => FALSE,
+    '#field_prefix' => t('Visitor must have '),
+    '#field_suffix' => t(' of the specified contexts'),
+    '#description' => t('Choose how multiple contexts should be applied to options. Choose "any" if the rule should apply if the user has any of the contexts. Choose "all" if the rule should apply only if the user has all of the contexts.'),
+    '#options' => array(
+      'OR' => 'any',
+      'AND' => 'all'
+    ),
+    '#default_value' => $default_strategy,
+  );
+  return $form;
+}
```

Comment:
I was keeping them for reference until the end of the who/what form stuff.  Do you think I should remove as I go to keep relevant things together?  I guess I can always look back in the code history.